app/controllers/users/noticed_notifications_controller.rb

```diff
@@ -11,8 +11,15 @@ def index
   def mark_as_read
     notification = NoticedNotification.find(params[:notification_id])
     notification.update(read_at: Time.zone.now)
-    project = notification.params[:project]
-    redirect_to user_project_path(project.author, project)
+    case NoticedNotificationType.new(params).notify
+    when true
+      assignment = notification.params[:assignment]
+      group = assignment.group
+      redirect_to group_assignment_path(group, assignment)
+    when false
+      project = notification.params[:project]
+      redirect_to user_project_path(project.author, project)
+    end
```

Comment:
Create a service which returns redirect path and success, you can have something like this. 
```
class NotifyUser
  Result = Struct.new(:success, :redirection_path)
  def initialize(notification_type, notifiable)
  end
  
  def call
    return Struct.new(,)
  end
end
```